### PR TITLE
sokol: init at unstable-2022-06-13

### PR DESCRIPTION
**Pull request by jonnynightingale**
###### Description of changes

Sokol is a popular (3.6k stars on Github) set of minimal C headers. Since this is a header-only library, the install process does not require any compilation.

https://github.com/floooh/sokol

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6303,6 +6303,12 @@
     githubId = 1843676;
     name = "Jonathan Reeve";
   };
+  jonnybolton = {
+    email = "jonnybolton@gmail.com";
+    github = "jonnybolton";
+    githubId = 8580434;
+    name = "Jonny Bolton";
+  };
```
Comment (legendofmiracles) [inline maintainers/maintainer-list.nix:6311]: Different commit: `maintainers: add jonnybolton`

Comment (jonnybolton) [reply maintainers/maintainer-list.nix]: I have moved the change to maintainer-list.nix into #177681

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -6303,12 +6303,6 @@
     githubId = 1843676;
     name = "Jonathan Reeve";
   };
-  jonnybolton = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix]: Split this commit into two.

If you need some tutorial, read this SO post:

https://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (AndersonTorres) [reply maintainers/maintainer-list.nix]: Don't do a _different pull request_, but a _different commit_.

Comment (jonnybolton) [reply maintainers/maintainer-list.nix:6311]: Apologies, I misunderstood. This should be correct now.

File: pkgs/development/libraries/sokol/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+stdenv.mkDerivation {
+  pname = "sokol";
+  version = "unstable-2022-06-13";
+
+  src = fetchFromGitHub {
+    owner = "floooh";
+    repo = "sokol";
+    rev = "3c7016105f3b7463f0cfc74df8a55642e5448c11";
+    sha256 = "1dkh65n3r1i97yr6b05prmg1s64yb23275v5p57s5rg9ckldp8bl";
```
Comment (AndersonTorres) [inline pkgs/development/libraries/sokol/default.nix:11]:
```suggestion
    hash = "sha256-dKHb6GTp5aJPuWWXI4ZYnhgdXs23gGWyPymGPGwxcLY=";
```

When possible, convert the SHA256 hashes to SRI format using the command 

`nix hash to-sri sha256:<hash>`

and pasting the result

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/sokol/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  version = "unstable-2022-06-13";
+
+  src = fetchFromGitHub {
+    owner = "floooh";
+    repo = "sokol";
+    rev = "3c7016105f3b7463f0cfc74df8a55642e5448c11";
+    sha256 = "sha256-dKHb6GTp5aJPuWWXI4ZYnhgdXs23gGWyPymGPGwxcLY=";
+  };
+
+  dontBuild = true;
```
Comment (AndersonTorres) [inline pkgs/development/libraries/sokol/default.nix:14]: If there is no `confugurePhase`, it is fine to add `dontConfigure = true;` too.

File: pkgs/development/libraries/sokol/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  installPhase = ''
```
Comment (AndersonTorres) [inline pkgs/development/libraries/sokol/default.nix:16]: Whenever using custom `...Phase`s, decorate them with `runHook`s:

```suggestion
  installPhase = ''
    runHook preInstall

```

File: pkgs/development/libraries/sokol/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    mkdir -p $out/include/sokol
+    cp *.h $out/include/sokol/
+    cp -R util $out/include/sokol/util
+  '';
```
Comment (AndersonTorres) [inline pkgs/development/libraries/sokol/default.nix:20]:
```suggestion

    runHook postInstall
  '';
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/sokol/default.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    repo = "sokol";
+    rev = "3c7016105f3b7463f0cfc74df8a55642e5448c11";
+    sha256 = "sha256-dKHb6GTp5aJPuWWXI4ZYnhgdXs23gGWyPymGPGwxcLY=";
+  };
+
+  dontBuild = true;
+  dontConfigure = true;
+
+  installPhase = ''
+    runHook preInstall
```
Comment (AndersonTorres) [inline pkgs/development/libraries/sokol/default.nix:18]:
```suggestion
```

File: pkgs/development/libraries/sokol/default.nix [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    repo = "sokol";
+    rev = "3c7016105f3b7463f0cfc74df8a55642e5448c11";
+    sha256 = "sha256-dKHb6GTp5aJPuWWXI4ZYnhgdXs23gGWyPymGPGwxcLY=";
+  };
+
+  dontBuild = true;
+  dontConfigure = true;
+
+  installPhase = ''
+
```
Comment (AndersonTorres) [inline pkgs/development/libraries/sokol/default.nix:18]:
```suggestion
```

**Comment by AndersonTorres** [comment]
Oh hell. Can you please force-push again?

**Pull request merged**